splicemachine/ml/management.py

```diff
@@ -1,33 +1,165 @@
+from base64 import b64encode as encode_base64
+from builtins import super
+from collections import defaultdict
+from os import environ as env_vars
+from sys import getsizeof
+from time import time, sleep
 import mlflow
-import mlflow.h2o
-import mlflow.sklearn
 import mlflow.spark
+import requests
 from mlflow.tracking import MlflowClient
+from py4j.java_gateway import java_import
+from pyspark.ml import PipelineModel
+from pyspark.ml.base import Model as SparkModel
+
+
+def get_pod_uri(pod, port, pod_count=0, testing=False):
+    """
+    Get address of MLFlow Container for Kubernetes
+    """
+
+    if testing:
+        return "http://{pod}:{port}".format(pod=pod, port=port)  # mlflow docker container endpoint
+
+    try:
+        return env_vars['MLFLOW_URL']
+    except KeyError as e:
+        raise KeyError(
+            "Uh Oh! MLFLOW_URL variable was not found... are you running in the Cloud service?")
+
+===
 from mlflow.exceptions import MlflowException
 
-def get_pod_uri(pod, port, pod_count=0):
-	import os
-	try:
-		return 'http://{pod}-{pod_count}-node.{framework}.mesos:{port}'.format(pod=pod, pod_count=pod_count, framework=os.environ['FRAMEWORK_NAME'], port=port)
-	except KeyError as e:
-		raise KeyError("Uh Oh! FRAMEWORK_NAME variable was not found... are you running in Zeppelin?")
+def _get_user():
+    """
+    Get the current logged in user to
+    Jupyter
+
+    :return: (str) name of the logged in user
+    """
+    try:
+        uname = env_vars['JUPYTERHUB_USER']
+        return uname
+    except KeyError:
+        raise Exception("Could not determine current running user. Running MLManager outside of Splice Machine Cloud Jupyter "
+                        "is currently unsupported")
+
+
+def _readable_pipeline_stage(pipeline_stage):
+    """
+    Get a readable version of the Pipeline stage
+    (without the memory address)
+    :param pipeline_stage: the name of the stage to parse
+    """
+    if '_' in str(pipeline_stage):
+        return str(pipeline_stage).split('_')[0]
+    return str(pipeline_stage)
+
+
+def _get_stages(pipeline):
+    """
+    Extract the stages from a fit or unfit pipeline
+
+    :param pipeline: a fit or unfit Spark pipeline
+    :return: stages list
+    """
+    if hasattr(pipeline, 'stages') and isinstance(pipeline.stages, list):
+        return pipeline.stages  # fit pipeline
+    return pipeline.getStages()  # unfit pipeline
+
+
+def _parse_string_parameters(string_parameters):
+    """
+    Parse string rendering of extractParamMap
+    :param string_parameters:
+    :return:
+    """
+    parameters = {}
+    parsed_mapping = str(string_parameters).replace("{", "").replace(
```

Comment:
Looks like you are trying to parse a json object that is represented as a string.  Is there a more native way to do this?